src/services/userService.js

```diff
@@ -0,0 +1,44 @@
+const { data } = require('../data');
+
+const getAll = () => {
+  return data.users;
+};
+
+const getById = (id) => {
+  return data.users.find((user) => user.id === id) || null;
+};
+
+const updateUser = ({ id, name }) => {
+  const user = getById(id);
+
+  Object.assign(user, { name });
+
+  return user;
+};
```

Comment:
In the updateUser function, you are not handling the case where a user with the given id does not exist. If getById(id) returns null, Object.assign will throw a TypeError. You should add a check for this and handle it appropriately.